### PR TITLE
Add chromedash-form-table and -form-field and use for all feature field forms

### PR DESCRIPTION
**Pull request by dlaliberte**
This PR replaces all feature form tables with chromedash-form-table and all rows with chromedash-form-field.  

Since tables and trs depend on each other, initial dom parsing will remove all content of tables that is not trs, so it is important to replace all the tables that contain any chromedash-form-field, and thus all trs also must be replaced by chromedash-form-field.  This applies to guide/{new.html, metadata.html, stage.html, editall.html}.

The resulting layout is the same as before, except for a few cases where we were explicitly using colspan="2" in manually generated table rows.  e.g. the new.html radio button group for feature type.

image

### CODE REVIEW

**Comment by dlaliberte** [comment]
image

**Comment by jrobbins** [comment]
I see what you are doing, but I'm not 100% clear on why we need these custom elements instead of using standard HTML `<table>` elements directly.    This PR description needs a sentence or two about the problem that you are trying to solve.

**Comment by dlaliberte** [comment]
> I see what you are doing, but I'm not 100% clear on why we need these custom elements instead of using standard HTML `<table>` elements directly. This PR description needs a sentence or two about the problem that you are trying to solve.

The second paragraph of the description explains why we need components for both the row groups and the containing table:  `Since tables and trs depend on each other, initial dom parsing will remove all content of tables that is not trs, ...`

I initially tried only defining the chromedash-form-field, but then found that the dom parsing moved them out of the table container to be after (or before?) the table, so those rows were displayed out of order, and styling was inconsistent.  Maybe if I add style="display:table-row-group" to each chromedash-form-field, they will behave like rows and stay where they were defined.  

But furthermore, defining a custom element for the table (chromedash-form-table) and replacing all the table rows with chromedash-form-field allows us to more easily change the layout of all the tables and rows.  For example, we can add more columns across all the rows, or use a grid layout instead of table layout.

File: pages/guideforms.py [hunk 1]
```diff
@@ -832,12 +832,17 @@ class ChromedashForm(forms.Form):
 
     def as_table(self):
         "Return this form rendered as HTML <tr>s -- excluding the <table></table>."
-        header = '<tr%(html_class_attr)s><th colspan="2"><b>%(label)s</b></th></tr>'
-        html = header + '<tr%(html_class_attr)s><td>%(errors)s%(field)s</td><td>%(help_text)s</td></tr>'
+        # header = '<tr%(html_class_attr)s><th colspan="2"><b>%(label)s</b></th></tr>'
```
Comment (jrobbins) [inline pages/guideforms.py:835]: Unneeded lines should be deleted.

File: static/elements/chromedash-form-table.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import {LitElement, css, html} from 'lit';
+// import {ifDefined} from 'lit/directives/if-defined.js';
```
Comment (jrobbins) [inline static/elements/chromedash-form-table.js:2]: Delete if not needed.

**Review by jrobbins — APPROVED** [review]
LGTM after addressing some comments.  I think I put in the same comments on an earlier version but didn't send them, so they may be duplicated.

File: static/elements/chromedash-form-table.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlaliberte) [reply static/elements/chromedash-form-table.js:2]: Deleting SHARED_STYLES too.

File: pages/guideforms.py (hunk 1 above)
Comment (dlaliberte) [reply pages/guideforms.py:835]: done.

File: static/elements/chromedash-form-field.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import {LitElement, css, html} from 'lit';
+import {ifDefined} from 'lit/directives/if-defined.js';
+// import {live} from 'lit/directives/live.js';
```
[the remark this answers is not in the document]
Comment (dlaliberte) [reply static/elements/chromedash-form-field.js]: done.  also removed SHARED_STYLES.

File: static/elements/chromedash-form-table.js (hunk 2 above)
Comment (dlaliberte) [reply static/elements/chromedash-form-table.js:2]: done

**Pull request merged**